PowerNSX.psm1

```diff
@@ -21191,6 +21191,432 @@ function Remove-NsxSecurityGroupMember {
+            }
+            elseif ( $entity -is [VMware.VimAutomation.ViCore.Interop.V1.VirtualDevice.NetworkAdapterInterop] ) {
+                #See NSX API guide 'Attach or Detach a Virtual Machine from a Logical Switch' for
+                #how to construct NIC id.
+                $vmUuid = ($entity.parent | get-view).config.instanceuuid
+                $EntityObjectId = "$vmUuid.$($entity.id.substring($entity.id.length-3))"
+
+            }
+            elseif (( $entity -is [VMware.VimAutomation.ViCore.Interop.V1.VIObjectInterop]) -and ( $NsxMemberTypes -contains $entity.ExtensionData.MoRef.Type)) {
+                $EntityObjectId = $entity.ExtensionData.MoRef.Value
+            }
+            else {
+                throw "Invalid member specified $($entity)"
+            }
+
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "key" -xmlElementText "ENTITY"
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "criteria" -xmlElementText "belongs_to"
+            # Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "value" -xmlElementText $entity.objectId
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "value" -xmlElementText $EntityObjectId
+        }
+        elseif ($PSCmdlet.ParameterSetName -eq "search") {
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "key" -xmlElementText $key.ToUpper()
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "criteria" -xmlElementText $criteria.ToLower()
+            Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "value" -xmlElementText $value
+        }
+
+        $xmlDynamicCriteria
+
+    }
+
+    end{}
+}
+
+function Add-NsxDymanicMemberSet {
+
+    <#
+    .SYNOPSIS
+    Adds a new dynamic member set to an existing NSX Security Group.
+
+    .DESCRIPTION
+    An NSX Security Group is a grouping construct that provides a powerful
+    grouping function that can be used in DFW Firewall Rules and the NSX
+    Service Composer.
+
+    This cmdlet adds a new dynamic member set to an existing NSX Security Group.
+
+    A Security Group can consist of Static Includes and Excludes as well as
+    dynamic matching properties.
+
+    A dynamic member set defines the criteria that an object must meet for it to
+    be added to the security group. This gives you the ability to include
+    virtual machines by defining a filter criteria with a number of parameters
+    supported to match the search criteria.
+
+    A Security Group can have multiple dynamic member sets in an logical AND/OR
+    arrangement, and within each of the dynamic member sets, the set may contain
+    multiple dynamic criteria in an AND/OR arrangement.
+
+    .EXAMPLE
+
+    $criteria1Spec = New-NsxDynamicCriteriaSpec -key VM.name -operator AND -criteria contains -value "PROD"
+
+    $criteria2Spec = New-NsxDynamicCriteriaSpec -key VM.GUEST_OS_FULL_NAME -operator AND -criteria contains -value "Win"
+
```

Comment:
ditto... you get the idea. :)  For the rest of the PR - any empty line that doesn't separate two logically different peices of code should be killed.  Especially, kill empty lines between code and closing braces.